spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/it/DatastoreIntegrationTests.java

```diff
@@ -718,6 +718,20 @@ public void readOnlyCountTest() {
 		assertThat(this.transactionalTemplateService.findByIdInReadOnly(1)).isEqualTo(testEntityA);
 	}
 
+	@Test
+	public void sameClassDescendantsTest() {
+		Student entity3 = new Student(Collections.EMPTY_LIST);
+		Student entity2 = new Student(Collections.EMPTY_LIST);
+		Student entity1 = new Student(Arrays.asList(entity2, entity3));
+		School school = new School(1L, Arrays.asList(entity1));
+		this.datastoreTemplate.save(school);
+
+		School readSchool = this.datastoreTemplate.findById(school.id, School.class);
+		Student child = readSchool.descendants.get(0);
```

Comment:
add assertion to verify that `readSchool.descendants` contain only `entity1`

Reply:
Done.